tensorflow/compiler/xla/service/gpu/gpu_fusible.cc

```diff
@@ -131,5 +131,41 @@ bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
                                              get_loop_shape(instr_2));
 }
 
+bool IsInputFusible(const HloInstruction& instr) {
+  // Input fusion only handles non-elemental reduction and scatter operations.
+  return IsInputFusibleReduction(instr) ||
+         instr.opcode() == HloOpcode::kScatter;
```

Comment:
This breaks InstructionFusionTest.FuseIntoScatter. The function works for unfused Scatter ops but not for existing Scatter input fusions. You want to add something like this.

         (instr.opcode() == HloOpcode::kFusion &&
          instr.fusion_kind() == HloInstruction::FusionKind::kInput &&
          instr.fused_expression_root()->opcode() == HloOpcode::kScatter)